File: guides/spring-boot-to-micronaut-actuator/micronautframework/java/src/test/java/example/micronaut/HealthTest.java
package example.micronaut;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpStatus;
import io.micronaut.http.client.BlockingHttpClient;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;
import java.util.Optional;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
class HealthTest {
    @Test
    void healthEndpointReturnsOkAndStatusUp(@Client("/") HttpClient httpClient) {
        BlockingHttpClient client = httpClient.toBlocking();
        String url = "/actuator/health";
        HttpResponse<String> response = client.exchange(url, String.class);
        assertEquals(HttpStatus.OK, response.getStatus());
        Optional<String> jsonOptional = response.getBody();
        assertTrue(jsonOptional.isPresent());
        String json = jsonOptional.get();
        assertNotNull(json);
        assertEquals("{\"status\":\"UP\"}", json);
    }
}

File: guides/micronaut-health-endpoint/kotlin/src/test/kotlin/example/micronaut/HealthTest.kt
package example.micronaut

import io.micronaut.http.HttpRequest
import io.micronaut.http.HttpStatus
import io.micronaut.http.client.HttpClient
import io.micronaut.http.client.annotation.Client
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@MicronautTest  // <1>
class HealthTest(@Client("/") val client: HttpClient) { // <2>

    @Test
    fun healthEndpointExposed() {
        val status = client.toBlocking().retrieve(HttpRequest.GET<Any>("/health"), HttpStatus::class.java) // <3>
        assertEquals(HttpStatus.OK, status)
    }
}